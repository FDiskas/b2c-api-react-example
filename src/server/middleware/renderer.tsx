import React from 'react';
import ReactDOMServer from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import serialize from 'serialize-javascript';

import { App } from 'src';
import { configureStore } from '@stores/configureStore';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import { Response as ExResponse, Request } from 'express';

const path = require('path');
const APP_TITLE = process.env.APP_TITLE || 'App';
const store = configureStore(createMemoryHistory());

export default (req: Request, res: ExResponse) => {
  const filePath = path.resolve(__dirname, '..', '..', 'index');
  const context: Partial<Response> = {};

  if (context.url) {
    res.redirect(302, context.url);
  } else {
    const status = context.status || 200;
    const html = ReactDOMServer.renderToString(
      <Provider store={store}>
        <StaticRouter location={req.url} context={context}>
          <App />
        </StaticRouter>
      </Provider>,
    );

    res.render(
      filePath,
      {
        state: serialize(store.getState()),
        title: `${APP_TITLE} SSR`,
      },
      (err: Error, htmlData: String) => {
        if (err) {
          console.error('err', err);

          return res.status(404).end();
        }

        return (
          res
            .status(status)
            .type('html')
            // inject the rendered app into our html and send it
            .end(htmlData.replace('<div id="app"></div>', `<div id="app">${html}</div>`))
        );
      },
    );
  }
};
