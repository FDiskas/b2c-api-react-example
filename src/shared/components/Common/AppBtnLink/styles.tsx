import { Theme } from '@material-ui/core/styles/createMuiTheme';
import createStyles from '@material-ui/core/styles/createStyles';

export const styles = (theme: Theme) => createStyles({
  root: {
    display: "inline-flex",
    border: `1px solid ${theme.appColors.black}`,
    textDecoration: "none",
    backgroundColor: theme.appColors.white,
    color: theme.appColors.black,
    lineHeight: "normal",
    letterSpacing: 1.5,
    fontSize: theme.appFixedDimensions.fontSize.medium,
    fontWeight: 500,
    borderRadius: theme.appFixedDimensions.borderRadius,
    paddingRight: theme.spacing.unit * 6,
    paddingLeft: theme.spacing.unit * 6,
    paddingTop: theme.spacing.unit * 1.5,
    paddingBottom: theme.spacing.unit * 1.5,
    transition: "all 0.4s ease",
    textAlign: "center",
    [theme.breakpoints.down('xs')]: {
      display: "flex",
      justifyContent: "center",
    },

    "&:hover": {
      backgroundColor: theme.appColors.black,
      color: theme.appColors.white,
      transition: "all 0.4s ease",
    }
  },
});
