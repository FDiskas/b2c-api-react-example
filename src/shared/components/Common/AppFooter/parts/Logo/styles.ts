import createStyles from '@material-ui/core/styles/createStyles';

export const styles = () => createStyles({
  logoContainer: {
    display: 'flex',
    alignItems: 'center',
  },
  logo: {
    display: 'block',
    marginRight: 8,
  },
  logoCopy: {
    color: '#111',
    fontSize: '14px',
    lineHeight: '18px',
    opacity: .4
  }
});
