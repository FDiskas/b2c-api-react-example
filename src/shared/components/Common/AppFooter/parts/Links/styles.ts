import { Theme } from '@material-ui/core/styles/createMuiTheme';
import createStyles from '@material-ui/core/styles/createStyles';

export const styles = (theme: Theme) => createStyles({
  title: {
    textTransform: 'uppercase',
    margin: 0,
  },
  linkList: {
    listStyle: 'none',
    margin: 0,
    padding: 0,
  },
  linkItem: {},
  link: {
    textDecoration: 'none',
    color: '#000',
  },
});
