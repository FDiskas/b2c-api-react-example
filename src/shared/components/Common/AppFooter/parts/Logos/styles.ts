import { Theme } from '@material-ui/core/styles/createMuiTheme';
import createStyles from '@material-ui/core/styles/createStyles';

export const styles = (theme: Theme) => createStyles({
  logosList: {
    display: 'flex',
    alignItems: 'center',
    listStyle: 'none',
    margin: 0,
    padding: 0,
  },
  logosListItem: {
    margin: '0 14px',
  }
});
