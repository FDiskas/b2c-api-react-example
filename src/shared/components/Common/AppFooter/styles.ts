import { Theme } from '@material-ui/core/styles/createMuiTheme';
import createStyles from '@material-ui/core/styles/createStyles';

export const styles = (theme: Theme) => createStyles({
  footer: {
    background: theme.appColors.lightGrey,
    padding: '64px 0',
  },
  footerContainer: {
    ...theme.appContainerStyles,
  }
});
