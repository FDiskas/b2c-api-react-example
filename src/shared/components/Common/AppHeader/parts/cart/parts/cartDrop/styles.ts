import { Theme } from '@material-ui/core/styles/createMuiTheme';
import createStyles from '@material-ui/core/styles/createStyles';

export const styles = (theme: Theme) => createStyles({
  cartDrop: {
    width: 418,
    padding: 24,
    border: '1px solid #d8d8d8',
    boxShadow: '0 5px 10px 0 rgba(0, 0, 0, 0.05)',
  },
  title: {
    fontSize: '20px',
    lineHeight: '24px',
    fontWeight: 'bold',
    color: theme.appColors.black,
    letterSpacing: '-.5px',
    margin: '0 0 10px',
    borderBottom: '1px solid #d8d8d8',
    paddingBottom: 10,
    '& strong': {
      fontWeight: 400,
    },
  },
  cartDropProductsList: {
    listStyle: 'none',
    margin: 0,
    padding: 0,
    fontSize: 14,
    lineHeight: '18px',
    '& li': {
      borderBottom: '1px solid #d8d8d8',
    },
    '& a': {
      color: theme.appColors.black,
      textDecoration: 'none',
      display: 'block',
      padding: '24px 0',
    },
  },
  cartTotalContainer: {
    padding: '16px 0',
    borderBottom: '1px solid #d8d8d8',
    marginBottom: 24,
  },
  cartTotal: {
    fontSize: 16,
    lineHeight: '20px',
    fontWeight: 'bold',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    textTransform: 'uppercase',
    '&:not(:last-child)': {
      marginBottom: 10,
    },
  },
  priceTotal: {
    fontSize: 16,
    lineHeight: '20px',
    fontWeight: 'bold',
  },
  cartBtns: {
    display: 'flex',
    '& a': {
      flex: 1,
      '&:not(:last-child)': {
        marginRight: 12,
      },
    },
  },
});
