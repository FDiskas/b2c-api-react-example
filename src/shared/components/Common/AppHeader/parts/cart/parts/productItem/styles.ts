import { Theme } from '@material-ui/core/styles/createMuiTheme';
import createStyles from '@material-ui/core/styles/createStyles';

export const styles = (theme: Theme) => createStyles({
  productItem: {
    display: 'flex',
    alignItems: 'stretch'
  },
  image: {
    width: 100,
    display: 'flex',
    alignItems: 'center',
    '& img': {
      display: 'block',
      width: '100%',
      maxHeight: '100%',
    }
  },
  rowsContainer: {
    flexGrow: 1,
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    paddingLeft: 15,
  },
  topRow: {
    display: 'flex',
    justifyContent: 'space-between',
    marginBottom: 10,
  },
  name: {},
  price: {},
  bottomRow: {},
  quantity: {},
  btnRemove: {}
});
