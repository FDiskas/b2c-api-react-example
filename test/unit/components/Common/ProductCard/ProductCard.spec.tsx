import * as React from 'react';
import { HTMLAttributes, shallow, ShallowWrapper, mount } from "enzyme";
import Card from '@material-ui/core/Card';
import CardMedia from '@material-ui/core/CardMedia';
import renderer from 'react-test-renderer';

import {getFormattedPrice} from '../../../../../src/shared/services/priceFormatter';
import {
  ProductCardBase
} from '../../../../../src/shared/components/Common/ProductCard';
import {styles} from '../../../../../src/shared/components/Common/ProductCard/styles';


// Required props
const abstract_sku = "191";
const abstract_name = "test_name";
const price = 22222;
const images = [
  {
    "external_url_small": "//images.icecat.biz/img/gallery_mediums/img_17681791_medium_1482143992_4607_19487.jpg",
    "external_url_large": "//images.icecat.biz/img/norm/high/17681791-4446.jpg"
  }
];
const currency = "EUR";

const dispatch = () => null;

const getShallowedComponent = () => (
  shallow(
    <ProductCardBase
      abstract_sku={abstract_sku}
      abstract_name={abstract_name}
      price={price}
      images={images}
      currency={currency}
      classes={styles}
    />
  )
);

describe('components->Common->ProductCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper =  getShallowedComponent();
  });

  it("renders the component", () => {
    expect(wrapper.find(Card)).toHaveLength(1);
  });

  it("renders the image block", () => {
    expect(wrapper.find(CardMedia)).toHaveLength(1);
  });

  it("renders the image", () => {
    expect(wrapper.find(CardMedia).dive().dive().find('img').prop('src')).toEqual(images[0].external_url_small);
  });

  it("renders the product Name", () => {
    expect(wrapper.find('[data-type="productName"]').dive().dive().text()).toEqual(abstract_name);
  });

  it("renders the price", () => {
    expect(wrapper.find('[data-type="priceToShow"]').dive().dive().text()).toEqual(getFormattedPrice(price, currency));
  });

});

describe('components->Common->ProductCard: snapshot', () => {

  it('renders correctly', () => {
    const tree = renderer
      .create(
        getShallowedComponent()
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if something is missed', () => {
    const tree = renderer
      .create(
        <ProductCardBase
          abstract_sku={null}
          abstract_name={null}
          price={null}
          images={null}
          currency={currency}
          classes={styles}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});
