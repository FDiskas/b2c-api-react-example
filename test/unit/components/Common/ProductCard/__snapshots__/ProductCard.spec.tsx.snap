// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components->Common->ProductCard: snapshot renders correctly 1`] = `
<div
  className="MuiPaper-root-02 MuiPaper-elevation8-012 MuiPaper-rounded-03 MuiCard-root-01"
>
  <button
    className="MuiButtonBase-root-032 MuiCardActionArea-root-029"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <img
      className="MuiCardMedia-root-035 MuiCardMedia-media-036"
      src="//images.icecat.biz/img/gallery_mediums/img_17681791_medium_1482143992_4607_19487.jpg"
      title="test_name"
    />
    <div
      className="MuiCardContent-root-037"
    >
      <h2
        className="MuiTypography-root-038 MuiTypography-title-044 MuiTypography-gutterBottom-055"
        data-type="productName"
      >
        test_name
      </h2>
      <h3
        className="MuiTypography-root-038 MuiTypography-subheading-045 MuiTypography-colorPrimary-058 MuiTypography-gutterBottom-055"
        data-type="priceToShow"
      >
        € 22,222.00
      </h3>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight-031"
    />
    <span
      className="MuiTouchRipple-root-063"
    />
  </button>
  <div
    className="MuiCardActions-root-070"
  >
    <button
      className="MuiButtonBase-root-032 MuiButton-root-072 MuiButton-text-074 MuiButton-textSecondary-076 MuiButton-flat-077 MuiButton-flatSecondary-079 MuiButton-sizeSmall-095 MuiCardActions-action-071"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-073"
      >
        Learn More
      </span>
      <span
        className="MuiTouchRipple-root-063"
      />
    </button>
  </div>
</div>
`;

exports[`components->Common->ProductCard: snapshot renders correctly if something is missed 1`] = `
<div
  className="MuiPaper-root-02 MuiPaper-elevation8-012 MuiPaper-rounded-03 MuiCard-root-01"
>
  <button
    className="MuiButtonBase-root-032 MuiCardActionArea-root-029"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <div
      className="MuiCardContent-root-037"
    >
      <h2
        className="MuiTypography-root-038 MuiTypography-title-044 MuiTypography-gutterBottom-055"
        data-type="productName"
      />
      <h3
        className="MuiTypography-root-038 MuiTypography-subheading-045 MuiTypography-colorPrimary-058 MuiTypography-gutterBottom-055"
        data-type="priceToShow"
      >
        € 0.00
      </h3>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight-031"
    />
    <span
      className="MuiTouchRipple-root-063"
    />
  </button>
  <div
    className="MuiCardActions-root-070"
  >
    <button
      className="MuiButtonBase-root-032 MuiButton-root-072 MuiButton-text-074 MuiButton-textSecondary-076 MuiButton-flat-077 MuiButton-flatSecondary-079 MuiButton-sizeSmall-095 MuiCardActions-action-071"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-073"
      >
        Learn More
      </span>
      <span
        className="MuiTouchRipple-root-063"
      />
    </button>
  </div>
</div>
`;
